src/openforms/registrations/contrib/json_dump/config.py

```diff
@@ -85,3 +95,4 @@ class JSONDumpOptions(TypedDict):
     variables: list[str]
     fixed_metadata_variables: list[str]
     additional_metadata_variables: list[str]
+    transform_to_list: NotRequired[list]
```

Comment:
```suggestion
    transform_to_list: list
```
`NotRequired` means that the key may be missing in the dictionary after deserialization, but because a `default=list` is specified, the key will always be present, so we can effectively mark it as required.